doctor/types.py

```diff
@@ -105,7 +105,7 @@ def __init__(self, *args, **kwargs):
                 '{} did not define a description attribute'.format(cls))
 
 
```

Comment:
Not subclassing `SuperType` was causing any types that subclassed this to not get passed to the logic function and not get documented in api docs.